rbx_types/src/attributes/writer.rs

```diff
@@ -0,0 +1,121 @@
+use super::*;
+use crate::{basic_types::*, variant::Variant};
+
+fn write_f32(bytes: &mut Vec<u8>, n: f32) {
+    bytes.append(&mut n.to_le_bytes().to_vec());
+}
+
+fn write_u32(bytes: &mut Vec<u8>, n: u32) {
+    bytes.append(&mut n.to_le_bytes().to_vec());
+}
+
+fn write_color3(bytes: &mut Vec<u8>, color: Color3) {
+    write_f32(bytes, color.r);
+    write_f32(bytes, color.g);
+    write_f32(bytes, color.b);
+}
+
+fn write_string(bytes: &mut Vec<u8>, string: &str) {
+    write_u32(bytes, string.len() as u32);
+    for byte in string.bytes() {
+        bytes.push(byte);
+    }
+}
+
+fn write_udim(bytes: &mut Vec<u8>, udim: UDim) {
+    write_f32(bytes, udim.scale);
+    bytes.append(&mut udim.offset.to_le_bytes().to_vec());
+}
+
+fn write_vector2(bytes: &mut Vec<u8>, vector2: Vector2) {
+    write_f32(bytes, vector2.x);
+    write_f32(bytes, vector2.y);
+}
+
+/// Writes the attribute property (AttributesSerialize) from a map of attribute names -> values.
+pub fn attributes_from_map<
+    K: Into<String>,
+    V: Into<Variant>,
+    I: Iterator<Item = (K, V)> + ExactSizeIterator,
+    M: IntoIterator<IntoIter = I, Item = (K, V)>,
+>(
```

Comment:
I'm iffy about this type signature, would like a second glance at it.

I want it to capture HashMaps, Vec<(K, V)>, and stuff like BTreeMap.